"""Generated protobuf code for Requirement Analysis Agent."""

from .requirement_analysis_pb2 import (
    AnalyzeRequest,
    AnalyzeResponse,
    ReanalyzeRequest,
    ExportRequest,
    ExportResponse,
    ForwardRequest,
    ForwardResponse,
    HealthCheckRequest,
    HealthCheckResponse,
    JiraStoryInput,
    FreeFormInput,
    TranscriptInput,
    AnalysisConfig,
    QualityScore,
    DimensionScore,
    ExtractedRequirement,
    RequirementStructure,
    Gap,
    ClarifyingQuestion,
    GeneratedAC,
    DomainValidation,
    EntityMapping,
    ApplicableRule,
    DomainWarning,
    AnalysisMetadata,
    AnsweredQuestion,
    ACDecision,
    TestCasesConfig,
    StructuredRequirement,
)
from .requirement_analysis_pb2_grpc import (
    RequirementAnalysisServiceStub,
    RequirementAnalysisServiceServicer,
    add_RequirementAnalysisServiceServicer_to_server,
)
from .ecommerce_domain_pb2 import (
    DomainContextRequest,
    DomainContextResponse,
    KnowledgeRequest,
    KnowledgeResponse,
    KnowledgeItem,
    EntityRequest,
    EntityResponse,
    Entity,
    EntityField,
    BusinessRule,
    EntityRelationship,
)
from .ecommerce_domain_pb2_grpc import (
    EcommerceDomainServiceStub,
)
from .test_cases_pb2 import (
    GenerateTestCasesRequest as TestCasesGenerateRequest,
    GenerateTestCasesResponse as TestCasesGenerateResponse,
    UserStoryInput as TestCasesUserStoryInput,
)
from .test_cases_pb2_grpc import (
    TestCasesServiceStub,
)

__all__ = [
    # Requirement Analysis Messages
    "AnalyzeRequest",
    "AnalyzeResponse",
    "ReanalyzeRequest",
    "ExportRequest",
    "ExportResponse",
    "ForwardRequest",
    "ForwardResponse",
    "HealthCheckRequest",
    "HealthCheckResponse",
    "JiraStoryInput",
    "FreeFormInput",
    "TranscriptInput",
    "AnalysisConfig",
    "QualityScore",
    "DimensionScore",
    "ExtractedRequirement",
    "RequirementStructure",
    "Gap",
    "ClarifyingQuestion",
    "GeneratedAC",
    "DomainValidation",
    "EntityMapping",
    "ApplicableRule",
    "DomainWarning",
    "AnalysisMetadata",
    "AnsweredQuestion",
    "ACDecision",
    "TestCasesConfig",
    "StructuredRequirement",
    # Requirement Analysis Service
    "RequirementAnalysisServiceStub",
    "RequirementAnalysisServiceServicer",
    "add_RequirementAnalysisServiceServicer_to_server",
    # eCommerce Domain
    "DomainContextRequest",
    "DomainContextResponse",
    "KnowledgeRequest",
    "KnowledgeResponse",
    "KnowledgeItem",
    "EntityRequest",
    "EntityResponse",
    "Entity",
    "EntityField",
    "BusinessRule",
    "EntityRelationship",
    "EcommerceDomainServiceStub",
    # Test Cases
    "TestCasesGenerateRequest",
    "TestCasesGenerateResponse",
    "TestCasesUserStoryInput",
    "TestCasesServiceStub",
]
